core/src/main/java/com/scalar/db/storage/cosmos/SelectStatementHandler.java

```diff
@@ -135,6 +147,46 @@ private String makeQueryWithCondition(
     return select.getSQL(ParamType.INLINED);
   }
 
+  private SelectJoinStep<org.jooq.Record> makeQueryWithProjections(
+      Selection selection, TableMetadata tableMetadata) {
+    if (selection.getProjections().isEmpty()) {
+      return DSL.using(SQLDialect.DEFAULT).select().from("Record r");
+    }
+
+    List<String> projectedFields = new ArrayList<>();
+
+    // To project the required columns, we build a JSON object with the same structure as the
+    // `Record.class`so that each field can be deserialized properly into a `Record.class` object.
+    // For example, the projected field "r.id" will be mapped to the `Record.id` attribute
+    projectedFields.add("r.id");
+    projectedFields.add("r.concatenatedPartitionKey");
+    projectedFields.add("r.partitionKey");
+    projectedFields.add("r.clusteringKey");
```

Comment:
Just a question. Do we need project these fields in Cosmos DB even if we don't specify partition key and clustering key as projections?

Reply:
For this PR, for the sake of being consistent with the other adapters (otherwise I would have needed to make an exception in the integration testing for Cosmos), I projected by default the partition key and clustering key even though I will need to update the logic again in another PR to not project by default the partition and clustering key for all adapters.